web/src/components/njwds-extended/FileInput.test.tsx

```diff
@@ -0,0 +1,88 @@
+import { getMergedConfig } from "@/contexts/configContext";
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+import { FileInput } from "./FileInput";
+
+describe("<FileInput />", () => {
+  let file: File;
+  const Config = getMergedConfig();
+
+  beforeEach(() => {
+    file = new File(["my cool file contents"], "cool.png", { type: "image/png" });
+  });
+
+  it("passes file to onchange handler", async () => {
+    const mockOnChange = jest.fn();
+    render(
+      <FileInput
+        errorMessageRequired="error-message-required"
+        helperText="input-label"
+        onChange={mockOnChange}
+        value={undefined}
+      />
+    );
+
+    const uploader = screen.getByTestId("file-input");
+
+    fireEvent.change(uploader, { target: { files: [file] } });
+
+    const base64Contents = Buffer.from("my cool file contents", "utf8").toString("base64");
+
+    await waitFor(() => {
+      expect(mockOnChange).toHaveBeenCalledWith({
+        fileType: "PNG",
+        sizeInBytes: file.size,
+        base64Contents,
+        filename: "cool.png",
+      });
+    });
+  });
+
+  it("refuses files larger than the provided max file size", async () => {
+    const mockOnChange = jest.fn();
+    const errorMessageFileSize = Config.formation.fields.foreignGoodStandingFile.errorMessageFileSize;
+
+    file = new File(["x".repeat(1048576 / 2)], "cool.png", { type: "image/png" }); // .5 MB file
```

Comment:
this would be an example where I recommend a variable named `const halfMegabyteFile = "x".repeat...` instead of using a comment, to have the code be "self-documenting"